src/Routing/Middleware/RoutingMiddleware.php

```diff
@@ -55,7 +56,14 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
                 $response->getHeaders()
             );
         }
+        $middleware = Router::getMatchingMiddleware($request->getUri()->getPath());
```

Comment:
I wonder if instead of calling this function statically, this class could get the topmost routing collection instance so that it could call the method directly on it.